src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -321,6 +323,101 @@ network::network(program::ptr program, uint16_t stream_id)
 network::network(program::ptr program, stream::ptr stream, uint16_t stream_id)
     : network(program, stream, false, stream_id == 0) {}
 
+network::network(cldnn::BinaryInputBuffer& ib, stream::ptr stream, engine& engine, uint16_t stream_id)
+    : _program(nullptr)
+    , _engine(ib.get_engine())
+    , _stream(stream)
+    , _memory_pool(new memory_pool(engine))
+    , _internal(false)
+    , _is_primary_stream(false)
+    , _reset_arguments(true) {
+    net_id += 1;
+
+    uint32_t prog_id;
+    std::vector<std::string> batch_header_str;
+    ib >> prog_id;
+    ib >> batch_header_str;
+    kernels_cache kernels_cache(get_engine(), prog_id, batch_header_str);
+    ib >> kernels_cache;
+
+    int num_data_nodes;
+    ib >> num_data_nodes;
+
+    _memory_pool->clear_pool_for_network(net_id);
```

Comment:
Why is it needed? new mem pool object is created above, so I think clear is redundant.

Reply:
https://github.com/openvinotoolkit/openvino/pull/13986/files#r1024741591
I removed it. Thank you.